homeassistant/components/advantage_air/cover.py

```diff
@@ -0,0 +1,154 @@
+"""Cover platform for Advantage Air integration."""
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    DEVICE_CLASS_DAMPER,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    CoverEntity,
+)
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import ADVANTAGE_AIR_STATE_CLOSE, ADVANTAGE_AIR_STATE_OPEN, DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up AdvantageAir cover platform."""
+
+    instance = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for ac_key in instance["coordinator"].data["aircons"]:
+        for zone_key in instance["coordinator"].data["aircons"][ac_key]["zones"]:
+            # Only add zone vent controls when zone in vent control mode.
+            if (
+                instance["coordinator"].data["aircons"][ac_key]["zones"][zone_key][
+                    "type"
+                ]
+                == 0
+            ):
+                entities.append(AdvantageAirZoneVent(instance, ac_key, zone_key))
+    async_add_entities(entities)
+
+
+class AdvantageAirZoneVent(CoordinatorEntity, CoverEntity):
+    """Advantage Air Cover Class."""
+
+    def __init__(self, instance, ac_key, zone_key):
+        """Initialize the Advantage Air Zone Vent cover entity."""
+        super().__init__(instance["coordinator"])
+        self.async_change = instance["async_change"]
+        self.ac_key = ac_key
+        self.zone_key = zone_key
+
+    @property
+    def _zone(self):
+        return self.coordinator.data["aircons"][self.ac_key]["zones"][self.zone_key]
+
+    @property
+    def name(self):
+        """Return the name."""
+        return f'{self._zone["name"]}'
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-{self.zone_key}'
+
+    @property
+    def device_class(self):
+        """Return the device class of the vent."""
+        return DEVICE_CLASS_DAMPER
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_OPEN | SUPPORT_CLOSE | SUPPORT_SET_POSITION
+
+    @property
+    def is_closed(self):
+        """Return if vent is fully closed."""
+        return self._zone["state"] == ADVANTAGE_AIR_STATE_CLOSE
+
+    @property
+    def is_opening(self):
+        """Platform is unaware of opening state."""
+        return False
+
+    @property
+    def is_closing(self):
+        """Platform is unaware of closing state."""
+        return False
```

Comment:
I'm not sure you should report anything if it is not supported

Reply:
Not sure why I thought this was required, I can see they are blank in the inherited class, so I have removed them.